File: src/oneShot/testScripters/countLines/cl.py
# countLines - Count lines in files.
import sys
sys.path.append("/cluster/home/jsp/prog/utillities/py_modules")
import psyco
psyco.full()

def usage():
    """explain usage and exit"""
    sys.exit("""countLines - count lines in files
usage:
    countLines files[s]""")

def countFile(file):
    """Count lines in one file"""
    count = 0;
    f = open(file)
    while 1:
        line = f.readline()
	if line == '':
	    break
	count += 1
    print "total: %d" % count

if (len(sys.argv) != 2):
   usage()
countFile(sys.argv[1])

File: src/oneShot/testScripters/countLines/cl.c
/* countLines - Count lines in files. */
#include <stdio.h>

void usage()
/* Explain usage and exit. */
{
fprintf(stderr,
  "cl - Count lines in files\n"
  "cl:\n"
  "   countLines file[s]\n"
  );
exit(1);
}

void countFile(char *file)
/* Count lines in one file. */
{
FILE *f = fopen(file, "r");
char buf[1024];
int count = 0;
while (fgets(buf, sizeof(buf), f) != NULL)
    ++count;
fclose(f);
printf("total: %ld\n", count);
}

int main(int argc, char *argv[])
/* Process command line. */
{
if (argc != 2)
    usage();
countFile(argv[1]);
return 0;
}
